curvine-server/src/master/job/job_runner.rs

```diff
@@ -195,36 +194,38 @@ impl LoadJobRunner {
         job.update_state(JobTaskState::Pending, "Assigning workers");
         let block_size = job.info.block_size;
 
-        let mut total_size = 0;
-        let mut stack = LinkedList::new();
-        let mut task_index = 0;
-        stack.push_back(source_status);
-        while let Some(status) = stack.pop_front() {
+        let (mut total_size, mut stack) = (0i64, vec![source_status]);
+        let mut visited_paths = FastHashSet::<String>::with_capacity(128);
+        let (mut task_index, mut dir_count, mut file_count) = (0usize, 0u32, 0u32);
+
+        while let Some(status) = stack.pop() {
             if status.is_dir {
-                let dir_path = Path::from_str(status.path)?;
-                let childs = ufs.list_status(&dir_path).await?;
-                for child in childs {
-                    stack.push_back(child);
+                dir_count += 1;
+                let normalized_path = Path::from_str(&status.path)?
+                    .normalize_uri()
+                    .unwrap_or_else(|| status.path.clone());
+
+                if !visited_paths.insert(normalized_path) {
```

Comment:
I think the core problem is why ufs returns the same path instead of judging it here.